src/txn/txn_rollback_to_stable.c

```diff
@@ -659,6 +659,11 @@ __wt_txn_rollback_to_stable(WT_SESSION_IMPL *session, const char *cfg[])
      */
     WT_RET(__wt_open_internal_session(S2C(session), "txn rollback_to_stable", true, 0, &session));
     ret = __txn_rollback_to_stable(session, cfg);
+    /*
+     * Forcibly log a checkpoint after rollback to stable to ensure that both in-memory and on-disk
+     * versions are same.
+     */
```

Comment:
Do we need to perform checkpoint at the end of rollback to stable?

Reply:
With durable history, as part of rollback to stable, we modify the pages on the disk also. The checkpoint is to ensure that both on-disk and in-memory versions are the same. With follow-on force checkpoint the on-disk checkpoint version has only the stable updates according to the stable timestamp.

Reply:
We need to be careful with when we are incrementing the max generation number. 
If it is getting incremented after rollback to stable then I think is fine. Else below scenario will have an issue:
Scenario:
----------
Base generation is 99.
Increment generation to 100.
start rollback to stable
{
Aborted the obsolete updates, newer than stable timestamp.
Move the update at stable timestamp to data store. This will copy the old txn id as well.
During checkpoint page is written back to disk. (page generation number will be greater than 100).
Read the same page back. txn id will not be ignored as generation is greater than original startup generation number.
}
Please check this case, and reply where we are incrementing the startup generation, before rollback to stable or after rollback to stable ?

Reply:
As discussed, performing a checkpoint as part of rollback doesn't have a problem. But we need to fix the update that is moved from history store to data store as not to transfer the invalid transaction id.